src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SNI/SNIMarsConnection.cs

```diff
@@ -219,228 +192,250 @@ public void HandleSendComplete(SNIPacket packet, uint sniErrorCode)
+                                            // do not return the _partial packet, the receiver is responsible for returning the 
+                                            // packet once it has been used because it can take sync and async paths from here
+                                            _partial = null;
+                                            SqlClientEventSource.Log.TrySNITraceEvent(nameof(SNIMarsConnection), EventType.INFO, "SMUX_DATA | MARS Session Id {0}, Current Session {1} completed receiving Data", args0: _lowerHandle?.ConnectionId, args1: _header?.sessionId);
+                                            break;
+
+                                        case SNISMUXFlags.SMUX_ACK:
+                                            ReturnPacket(_partial);
+                                            _partial = null;
+                                            try
+                                            {
+                                                session.HandleAck(_header.highwater);
+                                                SqlClientEventSource.Log.TrySNITraceEvent(nameof(SNIMarsConnection), EventType.INFO, "SMUX_ACK | MARS Session Id {0}, Current Session {1} handled ack", args0: _lowerHandle?.ConnectionId, args1: _header?.sessionId);
+                                            }
+                                            catch (Exception e)
+                                            {
+                                                SNICommon.ReportSNIError(SNIProviders.SMUX_PROV, SNICommon.InternalExceptionError, e);
+                                            }
+                                            break;
+
+                                        case SNISMUXFlags.SMUX_FIN:
+                                            ReturnPacket(_partial);
+                                            _partial = null;
+                                            _sessions.Remove(_header.sessionId);
+                                            SqlClientEventSource.Log.TrySNITraceEvent(nameof(SNIMarsConnection), EventType.INFO, "SMUX_FIN | MARS Session Id {0}, SMUX_FIN flag received, Current Header Session Id {1} removed", args0: _lowerHandle?.ConnectionId, args1: _header?.sessionId);
+                                            break;
+
+                                        default:
+                                            Debug.Fail("unknown smux packet flag");
+                                            break;
                                     }
 
+                                    // partial packet has been fully reconstructed and dispatched so change to header state 
+                                    // to be ready to decode another smux header
+                                    _headerCount = 0;
+                                    _state = State.Header;
+
+                                    if (packet==null || packet.DataLeft == 0)
```

Comment:
```suggestion
                                    if (packet == null || packet.DataLeft == 0)
```

Reply:
Done. I thought the editorconfig file used to catch this when auto formatting but it doesn't seem to anymore.